### PR TITLE
fix: add Grpc authentication to merge mining proxy (see issue #4587)

### LINKED ISSUE #4587: Add GRPC auth to merge mining proxy

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
It is desirable that the Merge mining proxy has a GRPC authenticated wallet client connection. 

Motivation and Context
---
Contrary to the Mining wallet client connection, the merge mining proxy does not have GRPC authentication currently. This issue aims to add it to GRPC auth in the merge mining proxy.

Fixes https://github.com/tari-project/tari/issues/4587.

How Has This Been Tested?
---
Existing unit tests

### CODE REVIEW

**Comment by jorgeantonio21** [comment]
Fixes #4587.

**Review by sdbondi — DISMISSED** [review]
Looks good - will give it a try

File: applications/tari_merge_mining_proxy/src/config.rs [hunk 1]
```diff
@@ -41,6 +42,8 @@ pub struct MergeMiningProxyConfig {
     pub base_node_grpc_address: Multiaddr,
     /// The Tari console wallet's GRPC address
     pub console_wallet_grpc_address: Multiaddr,
+    /// GRPC authentication for console wallet
+    pub console_wallet_grpc_authentication: GrpcAuthentication,
```
Comment (sdbondi) [inline applications/tari_merge_mining_proxy/src/config.rs:46]: Please update the `common/config/presets/f_merge_mining_proxy.toml` file with the new setting

File: applications/tari_merge_mining_proxy/src/main.rs [hunk 2]
```diff
@@ -57,6 +62,23 @@ use crate::{
 };
 const LOG_TARGET: &str = "tari_mm_proxy::proxy";
 
+type WalletGrpcClient = grpc::wallet_client::WalletClient<InterceptedService<Channel, ClientAuthenticationInterceptor>>;
```
Comment (sdbondi) [inline applications/tari_merge_mining_proxy/src/main.rs]: Make this pub(crate)

File: applications/tari_merge_mining_proxy/src/proxy.rs [hunk 3]
```diff
@@ -62,6 +62,9 @@ pub(crate) const MMPROXY_AUX_KEY_NAME: &str = "_aux";
 /// The identifier used to identify the tari aux chain data
 const TARI_CHAIN_ID: &str = "xtr";
 
+type WalletClient = grpc::wallet_client::WalletClient<
```
Comment (sdbondi) [inline applications/tari_merge_mining_proxy/src/proxy.rs]:
```suggestion
use crate::WalletGrpcClient;
```

File: applications/tari_merge_mining_proxy/src/proxy.rs [hunk 4]
```diff
@@ -154,7 +157,9 @@ struct InnerService {
     block_templates: BlockTemplateRepository,
     http_client: reqwest::Client,
     base_node_client: grpc::base_node_client::BaseNodeClient<tonic::transport::Channel>,
-    wallet_client: grpc::wallet_client::WalletClient<tonic::transport::Channel>,
+    wallet_client: grpc::wallet_client::WalletClient<
```
Comment (sdbondi) [inline applications/tari_merge_mining_proxy/src/proxy.rs]:
```suggestion
    wallet_client: WalletGrpcClient
```

File: common/config/presets/f_merge_mining_proxy.toml [hunk 5]
```diff
@@ -42,6 +42,8 @@ monerod_url = [# stagenet
 # The Tari console wallet's GRPC address. (default = "/ip4/127.0.0.1/tcp/18143")
 #console_wallet_grpc_address = "/ip4/127.0.0.1/tcp/18143"
 
+# GRPC authentication for the Tari console wallet (default = "none")
+
```
Comment (sdbondi) [inline common/config/presets/f_merge_mining_proxy.toml:47]: Missing  `#wallet_grpc_authentication = { username: "miner", password: "$argon..." }`

File: applications/tari_merge_mining_proxy/src/main.rs (hunk 2 above)
Comment (jorgeantonio21) [reply applications/tari_merge_mining_proxy/src/main.rs]: Much better, changes made

File: common/config/presets/f_merge_mining_proxy.toml (hunk 5 above)
Comment (jorgeantonio21) [reply common/config/presets/f_merge_mining_proxy.toml:47]: Added

**Review by sdbondi — APPROVED** [review]

**Pull request merged**